solutions/usaco-1228

```diff
@@ -0,0 +1,86 @@
+---
+id: usaco-1228
+source: USACO Bronze 2022 Open
+title: Counting Liars
+author: Chongtian Ma
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_prob2_bronze_open22.html)
+
+## Alternate Explanation
+
+An important observation is that if Bessie is at position $x$, then the number of cows lying are all the
+cows with position less than $x$ that says "L" plus all the cows with a position greater than $x$ that says
+"G".
+
+Given this, we can loop through every cow's position and compute the total number 
+of liars by using a forwards loop and a backwards loop.
+
+The answer is the minimum liars over all the positions we looped over.
+
+## Implementation
+
+**Time Complexity**: $\mathcal{O}(N \log N)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+int main() {
+	int n;
+	cin >> n;
+	
+	vector<pair<int, char>> cows(n);
+	for (int i = 0; i < n; i++) {
+		// The position is read into .first for sorting.
+		cin >> cows[i].second >> cows[i].first;
+	}
+
+	sort(cows.begin(), cows.end());
+	
+	/*
+	 * lying_left[i] stores the number of cows to the left of cow i
+	 * that must be lying given that Bessie is at the position of cow i.
+	 * 
+	 * lying_right stores the same thing, but does it so for the cows
+	 * to the *right* of i.
+	 */
+	vector<int> lying_left(n);
+	vector<int> lying_right(n);
+	
+	for (int i = 1; i < n; i++) {
+		// Add up all the cows that are already lying to the left of our position.
+		lying_left[i] += lying_left[i - 1];
+		
+		if (cows[i - 1].second == 'L' && cows[i].first > cows[i - 1].first) {
+			/*
+			 * If the cow before says our position is to the left
+			 * but their position is strictly less than our position,
+			 * they're lying.
+			*/
+			lying_left[i]++;
+		}
+	}
+	
+	// Fill up lying_right in much the same way.
```

Comment:
```suggestion
	// Fill up lying_right the same way.
```

Reply:
not really "the same way", there's some variations